### PR TITLE
refac: introducing CalculationOutputWriter class

### PR DESCRIPTION
**Pull request by JesperGomes**
## Description
Introducing a CalculationOutputWriter class, which is intended to be responsible of writing basis data and result dataframes to the file system

## References

* #199

### CODE REVIEW

File: source/databricks/package/result_writer.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from package.constants import Colname
+
+
+class ResultWriter:
+    def __init__(
+        self,
+        batch_id: str,
+        results_path: str,
+    ):
+        self.batch_directory = f"{results_path}/batch_id={batch_id}"
```
Comment (BjarkeMeier) [inline source/databricks/package/result_writer.py:25]: Do or should we have tests ensuring that this directory is aligned with the .NET code?

File: source/databricks/package/result_writer.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    def write_basis_data_to_file(
```
Comment (BjarkeMeier) [inline source/databricks/package/result_writer.py:27]: Not sure, but how about either `write_basis_data` or `write_basis_data_to_files`?

**Review by BjarkeMeier — APPROVED** [review]
Nice, this is clearly an improvement of readability and maintainability.

File: source/databricks/package/result_writer.py (hunk 2 above)
Comment (JesperGomes) [reply source/databricks/package/result_writer.py:27]: Let's go with write_basis_data. Thanks for suggesting!

**Review by BjarkeMeier — APPROVED** [review]
I've updated your title and description according to the new writer class name.

**Pull request merged**